tests/deployments_test.go

```diff
@@ -0,0 +1,103 @@
+//
+// DISCLAIMER
+//
+// Copyright 2018 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Jan Christoph Uhde <jan@uhdejc.com>
+//
+package tests
+
+import (
+	"context"
+	"testing"
+
+	"github.com/dchest/uniuri"
+
+	driver "github.com/arangodb/go-driver"
+	api "github.com/arangodb/kube-arangodb/pkg/apis/deployment/v1alpha"
+	kubeArangoClient "github.com/arangodb/kube-arangodb/pkg/client"
+)
+
+// environements: provided from outside
+
+func TestDeploymentSingleMMFiles(t *testing.T) {
+	deploymentSubTest(t, api.DeploymentModeSingle, api.StorageEngineMMFiles)
+}
+
+func TestDeploymentSingleRocksDB(t *testing.T) {
+	deploymentSubTest(t, api.DeploymentModeSingle, api.StorageEngineRocksDB)
+}
+
+func TestDeploymentClusterMMFiles(t *testing.T) {
+	deploymentSubTest(t, api.DeploymentModeCluster, api.StorageEngineMMFiles)
+}
+
+func TestDeploymentClusterRocksDB(t *testing.T) {
+	deploymentSubTest(t, api.DeploymentModeCluster, api.StorageEngineRocksDB)
+}
+
+func deploymentSubTest(t *testing.T, mode api.DeploymentMode, engine api.StorageEngine) error {
+	// check environment
+	longOrSkip(t)
+
+	k8sNameSpace := getNamespace(t)
+	k8sClient := mustNewKubeClient(t)
+	deploymentClient := kubeArangoClient.MustNewInCluster() //
+
+	// Prepare deployment config
+	deploymentTemplate := newDeployment("test-1-deployment-" + string(mode) + "-" + string(engine) + "-" + uniuri.NewLen(4))
+	deploymentTemplate.Spec.Mode = mode
```

Comment:
Please merge in master.
After that, this code will no longer compile since `Mode` & `StorageEngine` are now references.
Fix it like this:
```
deploymentTemplate.Spec.Mode = api.NewMode(mode)
deploymentTemplate.Spec.StorageEngine = api.NewStorageEngine(engine)